### PR TITLE
pending votes implementation

### PR DESCRIPTION
**Pull request by haroldcarr**
Proofs/tests of this module will follow.
But I would like feedback and after incorporating feedback, to squash/commit this to main.

### CODE REVIEW

File: LibraBFT/Base/KVMap.agda [hunk 1]
```diff
@@ -59,8 +59,16 @@ module LibraBFT.Base.KVMap  where
...
+   kvm-insert-Haskell
+                  : (k : Key)(v : Val)(kvm : KVMap Key Val)
+                  → KVMap Key Val
+
+   kvm-member     : (k : Key) (kvm : KVMap Key Val)
+                  → Bool
+
    kvm-size       : KVMap Key Val → ℕ
    -- TODO-1: add properties relating kvm-toList to empty, kvm-insert and kvm-update
+   kvm-keys       : KVMap Key Val → List Key
```
Comment (cwjnkins) [inline LibraBFT/Base/KVMap.agda:71]: I suggest defining `kvm-keys` as `map proj1 . kvm-toList`

File: LibraBFT/Base/KVMap.agda [hunk 2]
```diff
@@ -59,8 +59,16 @@ module LibraBFT.Base.KVMap  where
                   → lookup k kvm ≢ nothing
                   → KVMap Key Val
 
+   kvm-insert-Haskell
+                  : (k : Key)(v : Val)(kvm : KVMap Key Val)
+                  → KVMap Key Val
+
+   kvm-member     : (k : Key) (kvm : KVMap Key Val)
```
Comment (cwjnkins) [inline LibraBFT/Base/KVMap.agda:66]: This could be defined as \ k -> is-just . lookup k

File: LibraBFT/Prelude.agda [hunk 3]
```diff
@@ -327,3 +327,10 @@ module LibraBFT.Prelude where
   f-sum f = sum ∘ List-map f
 
   open import LibraBFT.Base.Util public
+
+  -- HC-TODO : Lemmas has an isJust - should it be renamed, or this one?
+  isjust : ∀{ℓ}{A : Set ℓ} → Maybe A -> Bool
```
Comment (cwjnkins) [inline LibraBFT/Prelude.agda:332]: This is `is-just` in the Agda standard library.

Comment (haroldcarr) [reply LibraBFT/Prelude.agda:332]: I removed this def and used `is-just`.

File: LibraBFT/Base/KVMap.agda (hunk 1 above)
Comment (haroldcarr) [reply LibraBFT/Base/KVMap.agda:71]: I am going to deal with KVMap changes in an independent pull request.

File: LibraBFT/Base/KVMap.agda (hunk 2 above)
Comment (haroldcarr) [reply LibraBFT/Base/KVMap.agda:66]: I am going to deal with KVMap changes in an independent pull request.

**Pull request merged**